### PR TITLE
PHP 8 || Updated composer.json and PHPUnit

### PR DESCRIPTION
**Pull request by Moskito89**
Updating the PHP version constraint and PHPUnit allows the library to be installed under PHP 8 now

### CODE REVIEW

**Comment by Moskito89** [comment]
This also targets https://github.com/khanamiryan/php-qrcode-detector-decoder/issues/98

File: tests/QrReaderTest.php [hunk 1]
```diff
@@ -2,16 +2,17 @@
...
+class QrReaderTest extends TestCase
 {
 
     public function testText1()
     {
         $image = __DIR__ . "/qrcodes/hello_world.png";
 
         $qrcode = new QrReader($image);
-        $this->assertSame("Hello world!", $qrcode->text());
+        self::assertSame("Hello world!", $qrcode->text());
```
Comment (francislavoie) [inline tests/QrReaderTest.php:16]: This function isn't static, you should still use `$this->` 🤔

Comment (Moskito89) [reply tests/QrReaderTest.php:16]: Please take a look here: https://github.com/sebastianbergmann/phpunit/blob/1f0325a6d67a052df841de667ec6f1f3b19b3337/src/Framework/Assert.php#L1327 :-)

Comment (francislavoie) [reply tests/QrReaderTest.php:16]: Yes but that's just so you can call it statically if the method you're calling it from is also static. It's conventional to always use `$this->` for tests: https://phpunit.readthedocs.io/en/9.3/assertions.html#assertsame

Comment (Moskito89) [reply tests/QrReaderTest.php]: I found this article by Sebastian Bergmann where he writes, that there's no right or wrong about that: https://phpunit.de/manual/6.5/en/appendixes.assertions.html#appendixes.assertions.static-vs-non-static-usage-of-assertion-methods I prefer calling static methods statically, but I don't want to waste our time about that topic and changed it back. :-)

**Comment by Moskito89** [comment]
@francislavoie this is a small reminder for this PR. :-)

**Comment by francislavoie** [comment]
I don't own this repo 🤷‍♂️ I was just giving some feedback 😅

**Comment by Moskito89** [comment]
Damn, but thanks anyway! :-D @khanamiryan then this is up to you! :-)

**Comment by phpfui** [comment]
@Moskito89 Thanks for doing the PHP 8 port.  Only one comment, should the version for PHPUnit just be ">=5.7"?

Thinking that will allow it to work with older versions of PHP, as ^9.0 would seem to require PHPUnit 9, which will not work with older PHP versions.

Also might future proof the 8.1 upgrade, assuming the library does not have any 8.1 offending code.

**Comment by khanamiryan** [comment]
done

**Comment by francislavoie** [comment]
Thanks @khanamiryan! Please also tag a release with the change!

**Pull request merged**